### PR TITLE
Add checkstyle

### PR DESCRIPTION
**Pull request by HiranChaudhuri**
Automate enforcing of coding guidelines.

### CODE REVIEW

**Comment by HiranChaudhuri** [comment]
I am a bit surprised to see no checkstyle activity in the CI build. But then I realized I was testing locally like

`settlers-remake/jsettlers.tools$ ../gradlew clean build`

File: jsettlers.tools/build.gradle [hunk 1]
```diff
@@ -23,4 +24,12 @@ jar {
     }
 }
 
-tasks.run.workingDir = rootProject.projectDir
\ No newline at end of file
+tasks.run.workingDir = rootProject.projectDir
+
+project.extensions.findByName('checkstyle').sourceSets.each { SourceSet s ->
```
Comment (paulwedeck) [inline jsettlers.tools/build.gradle:29]: from what I can see the checkstyle file belongs to config/checkstyle/checkstyle.xml. Then it is no longer necessary to do the configuration stuff.
Also the plugin should probably be added to all submodules and not just jsettlers.tools

File: jsettlers.tools/gradle.properties [hunk 2]
```diff
@@ -0,0 +1,14 @@
+netbeans.org-netbeans-modules-editor-indent.text.x-java.CodeStyle.project.text-line-wrap=none
```
Comment (paulwedeck) [inline jsettlers.tools/gradle.properties:1]: If I am not mistaken no IDE except netbeans will try to interpret these lines. I don't want any IDE specific stuff in the git repo so please remove this

**Review by paulwedeck — CHANGES_REQUESTED** [review]
I tried to actually run one of these tests but there are countless warnings.
This was to be expected but it kind of questions the point of this pr.

File: jsettlers.tools/gradle.properties (hunk 2 above)
Comment (HiranChaudhuri) [reply jsettlers.tools/gradle.properties:1]: So far you eventually point out that tabs have to be used, or that the license header is not correct. I find it rather annoying to see such primitive things holding up either a quick merge or a discussion on the purpose of  a PR.

So the point is to agree and enforce on code quality, and I (and others) can directly get this feedback on every build rather than just before the merge. it will remove frustration on both sides.

The fact that it generates a lot of warnings shows that such code style was never enforced, so one could question why you are asking it in first place. But we are not just talking about formatting with 4 spaces or 8 or tabs. Checkstyle can check for so much more that makes the code easier readable, understandable, maintainable. 

So even if we do not agree on the concise configuration it currently has: let's activate some checks that we are sure to apply, and do others later if we feel like.

In short: Do not question the pull request. Come up with rules you feel comfortable about and activate them.

File: jsettlers.tools/build.gradle (hunk 1 above)
Comment (HiranChaudhuri) [reply jsettlers.tools/build.gradle:29]: I totally agree. But as I am not a gradle guru I was happy having added it to one project. Could you help move that stuff on a global level?

**Comment by HiranChaudhuri** [comment]
> I tried to actually run one of these tests but there are countless warnings.
> This was to be expected but it kind of questions the point of this pr.

Thinking further, this questions any of your requests to stick to some kind of code formatting.

File: jsettlers.tools/build.gradle (hunk 1 above)
Comment (paulwedeck) [reply jsettlers.tools/build.gradle:29]: Neither am I. I just did a quick google search and found [this stackoverflow question](https://stackoverflow.com/questions/26236308/how-to-apply-plugin-to-allprojects-with-new-gradle-plugins-mechanism)
